### PR TITLE
Truncate the table if the size exceeds more than 2GB

### PR DESCRIPTION
**Pull request by raghumdani**
## Summary

This change ensures the resultant record batches size does not exceed 2GB. Pyarrow uses int offsets and hence cannot hold arrays more than 2GB as the offset would overflow. 

## Rationale

The fix had already been applied but missed a corner case. 

## Changes

Address the corner case. 

## Impact

No impact on existing job runs. However, this change will allow onboarding tables with large parquet files.

## Testing

Added functional tests for both cases. 

## Regression Risk

Very Low

## Checklist

- [x] Unit tests covering the changes have been added
  - [x] If this is a bugfix, regression tests have been added

- [x] E2E testing has been performed

## Additional Notes

Any additional information or context relevant to this PR.

### CODE REVIEW

**Review by pfaraone — APPROVED** [review]
LGTM!

**Review by Zyiqin-Miranda — APPROVED** [review]
LGTM! Easier fix than I thought!

File: deltacat/tests/compute/compactor_v2/utils/test_primary_key_index.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_when_record_batches_exceed_int_max_size(self):
+        record = pa.array(["12bytestring" * 90_000_000])
+        record_batch = pa.RecordBatch.from_arrays([record], names=["pk"])
+        table = pa.Table.from_batches([record_batch, record_batch])
+
+        grouped_array = group_by_pk_hash_bucket(table, 3, ["pk"])
+
+        assert len(grouped_array) == 3
+        assert len(grouped_array[2].to_batches()) == 2  # two record batches preserved
```
Comment (Zyiqin-Miranda) [inline deltacat/tests/compute/compactor_v2/utils/test_primary_key_index.py:31]: Minor: Could you add doc here of why 2 record batches for maintenance? Also, should we also assert other two buckets have how many batches?

Comment (raghumdani) [reply deltacat/tests/compute/compactor_v2/utils/test_primary_key_index.py:31]: same pk, so other two buckets are empty.

Comment (Zyiqin-Miranda) [reply deltacat/tests/compute/compactor_v2/utils/test_primary_key_index.py:31]: Maybe just add a doc string on why two batches + other two buckets are empty just for future reference?

Comment (raghumdani) [reply deltacat/tests/compute/compactor_v2/utils/test_primary_key_index.py]: done.
